### PR TITLE
#167727318 Ft user should be able to view all his  trip request

### PR DESCRIPTION
**Pull request by solomonfrank**
#### What does this PR do?
endpoint that allow a logged in user to view all his pull request
#### Description of Task to be completed?
- ensure that the endpoint is working
- run a test on the endpoint on the endpoint
GET /trip
#### How should this be manually tested?
after cloning the repo, cd into it directory and Run 
- npm run migration
- npm db:seed to insert dummy data into the database
- npm run start:dev to start the server
use postman to test the endpoint by:
- POST /auth/siginin to login 
- copy return token and paste into the header in this format  Bearer Token
- test GET /trips
#### What are the relevant pivotal tracker stories?
#167727318

### CODE REVIEW

File: src/middlewares/authentication.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export const authenticateUserToken = (req, res, next) => {
   if (token && token.startsWith('Bearer ')) {
     token = token.slice(7, token.length);
   }
+  console.log(token)
```
Comment (joaquinto) [inline src/middlewares/authentication.js]: Nice work bro, please can you remove this changes.

File: src/helpers/responseHandler.js [hunk 2]
```diff
@@ -11,8 +11,11 @@
   * @param {Object} additionalFields
   * @returns {Object} null
   */
-export const respondWithSuccess = (res, statusCode = 200, message, additionalFields = {}) => res
-  .status(statusCode).send({ success: true, message, payload: { ...additionalFields } });
+export const respondWithSuccess = (res, statusCode = 200, message, additionalFields = {}) => {
+  additionalFields = Array.isArray(additionalFields) ? [ ...additionalFields ] : { ...additionalFields}
+  res.status(statusCode).send({ success: true, message, payload: additionalFields });
+}
```
Comment (joaquinto) [inline src/helpers/responseHandler.js:18]: Please, can you use the previous code and try to work around it. Thanks.

File: src/database/seeders/20190905042406-create-trip-request-seeder.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      returnDate: '1-Jan-2019',
+      userId: 1,
+      status:'approved',
+      createdAt: new Date(),
+      updatedAt : new Date()
+    },
+  ], {}),
+
+  down: (queryInterface, Sequelize) => queryInterface.bulkDelete('TripRequests', null, {})
+};
```
Comment (joaquinto) [inline src/database/seeders/20190905042406-create-trip-request-seeder.js]: Please, can you align your code well, you can do without `createdAt` and `updatedAt` in your seed data.

**Review by orley12 — COMMENTED** [review]
please implement @joaquinto feedback besides those LGTM

File: src/controllers/tripController.js [hunk 4]
```diff
@@ -20,6 +23,15 @@ export const oneWayTripRequest = async (req, res) => {
     return respondWithWarning(res, statusCode.internalServerError, 'Internal Server Error');
   }
 };
+export const viewAllTripRequest = async (req, res) => {
+  const { id } = req.auth;
+  try {
+    const tripRequests = await viewAllRequest(id);
```
Comment (Hector101) [inline src/controllers/tripController.js]: `viewAllRequest` can be renamed to `getTripRequests`

File: src/helpers/responseHandler.js (hunk 2 above)
Comment (Hector101) [reply src/helpers/responseHandler.js:18]: @joaquinto It's fine. this is account for when the payload is an array.

File: package.json [hunk 5]
```diff
@@ -4,20 +4,20 @@
   "description": "Making company travel and accomodation easy and convinient",
   "main": "index.js",
   "scripts": {
-    "test": "npm run migration:refresh:test && NODE_ENV=test npm run db:seed && NODE_ENV=test nyc --reporter=html --reporter=text mocha --timeout 20000 -r @babel/register ./src/test/**/*.test.js --exit",
+    "test": "cross-env npm run migration:refresh:test && cross-env NODE_ENV=test npm run db:seed && cross-env NODE_ENV=test nyc --reporter=html --reporter=text mocha --timeout 30000 -r @babel/register ./src/test/**/*.test.js --exit",
```
Comment (Hector101) [inline package.json]: why increase the timeout to `3000`?

File: src/controllers/tripController.js [hunk 6]
```diff
@@ -22,6 +23,15 @@ export const oneWayTripRequest = async (req, res) => {
     return respondWithWarning(res, statusCode.internalServerError, 'Internal Server Error');
   }
 };
+export const getTripRequests = async (req, res) => {
+  const { id } = req.auth;
+  try {
+    const tripRequests = await getRequest(id);
```
Comment (Hector101) [inline src/controllers/tripController.js]: `getRequest` can be renamed to `fetchTripRequests`.

**Pull request merged**